gensim/models/callbacks.py

```diff
@@ -0,0 +1,304 @@
+        Args:
+            corpus : Gensim document corpus.
+            texts : Tokenized texts. Needed for coherence models that use sliding window based probability estimator,
+                eg::
+                    texts = [['system', 'human', 'system', 'eps'],
+                             ['user', 'response', 'time'],
+                             ['trees'],
+                             ['graph', 'trees'],
+                             ['graph', 'minors', 'trees'],
+                             ['graph', 'minors', 'survey']]
+
+            dictionary : Gensim dictionary mapping of id word to create corpus. If model.id2word is present,
+                this is not needed. If both are provided, dictionary will be used.
+            window_size : Is the size of the window to be used for coherence measures using boolean sliding window as their
+                probability estimator. For 'u_mass' this doesn't matter.
+                If left 'None' the default window sizes are used which are:
+
+                    'c_v' : 110
+                    'c_uci' : 10
+                    'c_npmi' : 10
+
+            coherence : Coherence measure to be used. Supported values are:
+                'u_mass'
+                'c_v'
+                'c_uci' also popularly known as c_pmi
+                'c_npmi'
+                For 'u_mass' corpus should be provided. If texts is provided, it will be converted
+                to corpus using the dictionary. For 'c_v', 'c_uci' and 'c_npmi' texts should be provided.
+                Corpus is not needed.
+            topn : Integer corresponding to the number of top words to be extracted from each topic.
+            logger : Monitor training process using:
+                        "shell" : print coherence value in shell
+                        "visdom" : visualize coherence value with increasing epochs in Visdom visualization framework
+            viz_env : Visdom environment to use for plotting the graph
+            title : title of the graph plot
+        """
+        self.corpus = corpus
+        self.dictionary = dictionary
+        self.coherence = coherence
+        self.texts = texts
+        self.window_size = window_size
+        self.topn = topn
+        self.logger = logger
+        self.viz_env = viz_env
+        self.title = title
+
+    def get_value(self, **kwargs):
+        """
+        Args:
+            model : Pre-trained topic model. Should be provided if topics is not provided.
+                    Currently supports LdaModel, LdaMallet wrapper and LdaVowpalWabbit wrapper. Use 'topics'
+                    parameter to plug in an as yet unsupported model.
+            topics : List of tokenized topics.
+                eg::
+                    topics = [['human', 'machine', 'computer', 'interface'],
+                               ['graph', 'trees', 'binary', 'widths']]
+        """
+        # only one of the model or topic would be defined
+        self.model = None
```

Comment:
As both `model` and `topics` can be used to calculate Coherence, and only one of them would be defined in `**kwargs`. So this assignment is just to avoid name not defined error for the other variable which is not in `**kwargs`.